tests/test_ess_csc.py

```diff
@@ -671,3 +671,63 @@ async def test_lightning_data_client_nominal(self) -> None:
                 topic=self.remote.evt_sensorStatus,
                 sensorName="EssElectricField",
             )
+
+    async def test_weather_station_data_client_loses_connection(self) -> None:
```

Comment:
Thanks for adding this!

Is there any point to trying to test a working data client? I'd hate to duplicate much, if any, of the data client test code, so perhaps not.

Reply:
The advantage of adding a test for a working data client class would be to test the events and telemetry methods. Since that is done in ts_salobj already I wonder what the added value would be. Perhaps we can leave it as is for now and add it in case we ever see the need, for instance if and when we need to test specific data circumstances that cannot be tested by the DataClient alone.

Reply:
Sounds good to me. I like testing the data clients without having to create a CSC for them (though as we currently do it, I think we end up duplicating information in ts_xml -- something I think we can probably fix with Kafka salobj).